### PR TITLE
IGNITE-8154

### PR DESCRIPTION
**Pull request by antkr**

### CODE REVIEW

File: modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java [hunk 1]
```diff
@@ -499,6 +505,64 @@ public void testQueueMultipleThreads() throws Exception {
...
+            });
+
+            jmsStreamer.setDestination(dest);
+
+            try {
+                jmsStreamer.start();
+
+                fail("Expecting security exception");
+            } catch (IgniteException ie) {
+                assertTrue("Caught IgniteException, expected: ", ie.getCause().getCause() instanceof SecurityException);
```
Comment (slukyano) [inline modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java]: The message here is printed when the assertion fails, not passes, so it usually reads like "Unexpected exception...".

File: modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java [hunk 2]
```diff
@@ -499,6 +505,64 @@ public void testQueueMultipleThreads() throws Exception {
...
+            }
+
+            while(lsnrExceptions.isEmpty())
+                Thread.sleep(200);
+
+            assertTrue(lsnrExceptions.size() > 0);
+
+            try {
+                jmsStreamer.stop();
```
Comment (slukyano) [inline modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java]: Is it OK if the exception is NOT thrown (now the test will pass with either IgniteException or no exception)?

File: modules/jms11/src/main/java/org/apache/ignite/stream/jms11/JmsStreamer.java [hunk 3]
```diff
@@ -150,6 +152,9 @@
     /** Message listeners. */
     private Set<IgniteJmsMessageListener> listeners = Collections.newSetFromMap(new ConcurrentHashMap<IgniteJmsMessageListener, Boolean>());
 
+    /** The Jms {@link ExceptionListener} xception listener to use. */
```
[the remark this answers is not in the document]
Comment (antkr) [reply modules/jms11/src/main/java/org/apache/ignite/stream/jms11/JmsStreamer.java]: Done.

File: modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java [hunk 4]
```diff
@@ -499,6 +505,64 @@ public void testQueueMultipleThreads() throws Exception {
...
     }
 
+    /**
+     * Test for ExceptionListener functionality.
+     *
+     * @throws Exception If fails.
+     */
+    public void testExceptionListener() throws Exception {
+        // restart broker with auth plugin
+        if (broker.isStarted()) {
```
[the remark this answers is not in the document]
Comment (antkr) [reply modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java]: Done, missed this.

File: modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java (hunk 1 above)
Comment (antkr) [reply modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java]: Replaced with assertThrowsWithCause

File: modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java (hunk 2 above)
Comment (antkr) [reply modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java]: Done.

File: modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java [hunk 5]
```diff
@@ -537,29 +539,32 @@ public void testExceptionListener() throws Exception {
...
-            } catch (IgniteException ie) {
-                assertTrue("Caught IgniteException, expected: ", ie.getCause().getCause() instanceof SecurityException);
-            }
+                    return null;
+                }
+            }, SecurityException.class);
 
-            while(lsnrExceptions.isEmpty())
-                Thread.sleep(200);
+            latch.await(5, TimeUnit.SECONDS);
```
Comment (slukyano) [inline modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java]: Better to wrap `await()` calls into `assertTrue()` to make sure that the test fails if the `await()` times out.

Comment (antkr) [reply modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java]: Added that.

**Comment by slukyano** [comment]
Looks good to me.

**Pull request closed without merging**